src/components/grid/Column.js

```diff
@@ -12,82 +8,81 @@ const defaultShape = {
 
 const Column = React.createClass({
   propTypes: {
-    breakpoints: React.PropTypes.shape(defaultShape),
-    columnCount: React.PropTypes.number,
     offset: React.PropTypes.shape(defaultShape),
-    order: React.PropTypes.shape(defaultShape),
-    span: React.PropTypes.shape(defaultShape),
-    style: React.PropTypes.object
+    span: React.PropTypes.shape(defaultShape)
   },
 
   getDefaultProps () {
     return {
-      columnCount: 12,
-      order: {}
+      offset: { large: 0, medium: 0, small: 0 },
+      span: { large: 12, medium: 12, small: 12 }
     };
   },
 
-  getInitialState () {
-    return {
-      windowSize: this._getWindowSize()
-    };
-  },
+  getColumnWidths () {
+    const colWidths = [];
+    const small = this.props.span.small || 0;
+    const medium = this.props.span.medium || 0;
+    const large = this.props.span.large || 0;
+
+    // Column widths
+    if (small === 0) {
+      colWidths.push('hidden-sm');
+    } else {
+      colWidths.push('col-sm-' + small);
+    }
 
-  componentDidMount () {
-    window.addEventListener('resize', this._handleWindowResize);
-  },
+    if (medium === 0) {
+      colWidths.push('hidden-md');
+    } else if (medium !== small) {
+      colWidths.push('col-md-' + medium);
+    }
 
-  componentWillUnmount () {
-    window.removeEventListener('resize', this._handleWindowResize);
+    if (large === 0) {
+      colWidths.push('hidden-lg');
+    } else if (large !== medium) {
+      colWidths.push('col-lg-' + large);
+    }
+
+    return colWidths;
   },
 
-  _getWindowSize () {
-    const breakpoints = Object.assign({}, BreakPoints, this.props.breakpoints);
-    const width = window.innerWidth;
-    let windowSize = 'small';
+  getColumnOffsets () {
+    const offsets = [];
+    const small = this.props.offset.small || 0;
+    const medium = this.props.offset.medium || 0;
+    const large = this.props.offset.large || 0;
 
-    if (width >= breakpoints.large) {
-      windowSize = 'large';
-    } else if (width >= breakpoints.medium) {
-      windowSize = 'medium';
+    // Column offsets
+    if (small !== 0) {
+      offsets.push('col-sm-offset-' + small);
     }
 
-    return windowSize;
-  },
+    if (medium !== 0 && medium !== small) {
```

Comment:
It's the way the Bootstrap grid system works.  You don't add every class to the element if things are equal.  It just takes the one supplied value and uses it for small, medium, and large.  I believe it would most like work with out the check but then we would end up with a lot of classes when we don't have to.